### ISSUE TITLE
ISSUE: ONLY IN DEV BRANCH! UTF-8 not supported: command ENABLE illegal in state SELECTED, only allowed in states AUTH

### ISSUE THREAD

**Issue by Pr0mises** [moralmunky/Home-Assistant-Mail-And-Packages#815]
**Describe the bug**
In the dev branch the UTF8 support is not working properly with utf8 enabled subjects.

**Environment (please complete the following information):**
 - OS: Raspbian
 - Type of device: RPi4
 - Home Assistant version: 2023.4.4
 - Docker
 - Component version:-

**Logs**
2023-04-18 01:51:20.214 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] Amazon email search address: order-update@amazon.ca
2023-04-18 01:51:20.214 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] DEBUG subject: Livré
2023-04-18 01:51:20.214 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] DEBUG imap_search: SINCE 14-Apr-2023 SUBJECT
2023-04-18 01:51:20.215 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] UTF-8 not supported: command ENABLE illegal in state SELECTED, only allowed in states AUTH
2023-04-18 01:51:20.215 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] email_search value: ('BAD', 'command ENABLE illegal in state SELECTED, only allowed in states AUTH')
2023-04-18 01:51:20.215 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] Amazon email search address: order-update@amazon.co.uk
2023-04-18 01:51:20.216 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] DEBUG subject: Delivered: Your
2023-04-18 01:51:20.216 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] DEBUG imap_search: (FROM "order-update@amazon.co.uk" SUBJECT "Delivered: Your" SINCE 14-Apr-2023)
2023-04-18 01:51:20.238 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] email_search value: ('OK', [b''])
2023-04-18 01:51:20.239 DEBUG (SyncWorker_0) [custom_components.mail_and_packages.helpers] Amazon delivered email(s) found: 0

**Additional context**
The message accures because of  [`cmd = account.enable("UTF8=ACCEPT")`](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/ef90c58ad5ebfb993884b8cad4151f4a3b2706af/custom_components/mail_and_packages/helpers.py#L542 )

This call needs to be called before the selectfolder function and after the successful login [here](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/ef90c58ad5ebfb993884b8cad4151f4a3b2706af/custom_components/mail_and_packages/helpers.py#L162)

But even with that fix, the program couldn't find the email with the UTF8 enabled subject. I read alot about `SINCE <date> SUBJECT' but couldn't find any good resource about it other than https://datatracker.ietf.org/doc/html/rfc3501#page-52

Looking at this [stackoverflow](https://stackoverflow.com/questions/5640327/python-imap-search-using-a-subject-encoded-with-iso-8859-1) it looks like no one is actually using `account.enable("UTF8=ACCEPT")` so I deleted the `cmd = account.enable("UTF8=ACCEPT")` and reverted the [account.uid](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/ef90c58ad5ebfb993884b8cad4151f4a3b2706af/custom_components/mail_and_packages/helpers.py#L551) back to a `account.search()`
```
if utf8_flag:
  subject = subject.encode("utf-8")
  account.literal = subject
  try:
      # value = account.uid(
      #    "SEARCH", "CHARSET", "UTF-8", f"SINCE {date}", "SUBJECT"
      # )
      value = account.search("utf-8", "SUBJECT")
  except Exception as err:
      _LOGGER.debug(
          "Error searching emails with unicode characters: %s", str(err)
      )
      value = "BAD", err.args[0]
```

with that it's working and the subject get's parsed

**Comment by firstof9** [comment]
Feel free to make a PR with the fix.

**Comment by firstof9** [comment]
Fixed in #817
